discord/client.py

```diff
@@ -523,6 +525,48 @@ def tutorial(self) -> Tutorial:
         """
         return self._connection.tutorial
 
+    @property
+    def experiments(self) -> Sequence[UserExperiment]:
+        """Sequence[:class:`.UserExperiment`]: The experiments assignments for the connected client.
+
+        .. versionadded:: 2.1
+        """
+        return utils.SequenceProxy(self._connection.experiments.values())
+
+    @property
+    def guild_experiments(self) -> Sequence[GuildExperiment]:
+        """Sequence[:class:`.GuildExperiment`]: The guild experiments assignments for the connected client.
+
+        .. versionadded:: 2.1
+        """
+        return utils.SequenceProxy(self._connection.guild_experiments.values())
+
+    def get_experiment(self, experiment: Union[str, int], /) -> Optional[Union[UserExperiment, GuildExperiment]]:
+        """Returns a user or guild experiment from the given experiment identifier.
+
+        Parameters
+        -----------
+        experiment: Union[:class:`str`, :class:`int`]
+            The experiment name or hash to search for.
+
+        Returns
+        --------
+        Optional[Union[:class:`.UserExperiment`, :class:`.GuildExperiment`]]
+            The experiment, if found.
+        """
+        name = None
+        if not isinstance(experiment, int) and not experiment.isdigit():
+            name = experiment
+            experiment_hash = utils.murmurhash32(experiment, signed=False)
+        else:
+            experiment_hash = int(experiment)
+
+        exp = self._connection.experiments.get(experiment_hash, self._connection.guild_experiments.get(experiment_hash))
+        if exp and not exp.name and name:
+            # Backfill the name
+            exp.name = name
```

Comment:
Does this update the connection state array?

Reply:
It's the same object so yeah